File: API/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Microsoft.EntityFrameworkCore.Metadata.Internal;

namespace API.Entities
{
  public class Member
  {
    public string Id { get; set; } = null!; // This will be the same as AppUser.Id
    public DateOnly Birthday { get; set; }
    public string? ImageUrl { get; set; }
    public required string DisplayName { get; set; }
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime LastActive { get; set; } = DateTime.UtcNow;
    public required string Gender { get; set; }
    public string? Description { get; set; }
    public required string City { get; set; }
    public required string Country { get; set; }

    // Navigation Properties
    [JsonIgnore]
    [ForeignKey(nameof(Id))]
    public AppUser User { get; set; } = null!;
    [JsonIgnore]
    public List<Photo> Photos { get; set; } = [];
  }
}

File: client/src/types/member.ts
export interface Member {
  id: string;
  birthday: string;
  imageUrl?: string;
  displayName: string;
  created: string;
  lastActive: string;
  gender: string;
  description?: string;
  city: string;
  country: string;
}

export interface Photo {
  id: number;
  url: string;
  publicId?: string;
  memberId: string;
}

export type EditableMember = Pick<
  Member,
  "displayName" | "description" | "city" | "country"
>;

export class MemberParams {
  gender?: string;
  minAge = 18;
  maxAge = 120;
  pageNumber = 1;
  pageSize = 10;
  orderBy = 'age';
}
